sys/random/seed.c

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+ * @ingroup sys_random
+ * @{
+ * @file
+ *
+ * @brief PRNG seeding
+ *
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "log.h"
+#include "luid.h"
+#include "random.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+void auto_init_random(void)
+{
+    uint32_t seed;
+#ifdef FEATURE_PERIPH_CPUID
```

Comment:
Hmmm, this is not true e.g. for samr21-xpro or phytec boards. Currently figuring out why.

Reply:
With `FEATURES_OPTIONAL` in Makefile.dep, this flag is not set. I have no idea how to fix this in the build system.  Other alternatives:
1. Use `FEATURES REQUIRED` and live with a warning during compile time, for boards without this feature
2. Check for `CPUID_LEN`
3. Remove the `#ifdef` and rely on the backup ID behind the `LUID` module

The first proposal has been NACKed in a previous PR. The third proposal leverages improvements on the seeding function at proper code location. I would go for the second proposal.